cdap-docs/developers-manual/source/pipelines/developing-plugins/presentation-plugins.rst

```diff
@@ -0,0 +1,771 @@
+- :ref:`widget-type: <plugins-presentation-widgets>` The type of
+  widget to be used to represent this property.
+- **widget-attributes:** A map of attributes that the widget type requires to be defined in
+  order to render the property in the CDAP UI. The attributes required depend on the
+  widget type used.
+- :ref:`plugin-function: <plugins-presentation-plugin-function>`
+  An optional map of plugin method and its widget attributes that can be applied to a
+  particular plugin property.
+
+Note that with the exception of the value of the *label*, all properties and property
+values are case-sensitive.
+
+To find the available field names, you can use the :ref:`Artifact HTTP RESTful API 
+<http-restful-api-artifact>` to :ref:`retrieve plugin details 
+<http-restful-api-artifact-plugin-detail>` for an artifact, which will include all the
+available field names. (If the artifact is your own, you will already know the available
+field names from your source code.)
+
+In the case of a *Batch Source* plugin example, the ``configuration-groups`` could be
+represented by::
+
+  {
+    "configuration-groups": [
+      {
+        "label": "Batch Source",
+        "properties": [
+          {
+            "name": "name",
+            "label": "Dataset Name",
+            "widget-type": "dataset-selector"
+          },
+          {
+            "name": "basePath",
+            "label": "Dataset Base Path",
+            "widget-type": "textbox"
+          },
+          {
+            "name": "groupByFields",
+            "label": "Group By Fields",
+            "widget-type": "textbox",
+            "plugin-function": {
+              "method": "POST",
+              "widget": "outputSchema",
+              "output-property": "schema",
+              "plugin-method": "outputSchema",
+              "required-fields": ["groupByFields", "aggregates"],
+              "missing-required-fields-message":
+                "'Group By Fields' & 'Aggregates' properties are required to fetch schema."
+            }
+          },
+          {
+            "name": "duration",
+            "label": "Duration",
+            "widget-type": "textbox"
+          },
+          ...
+        ]
+      }
+    ]
+  }
+
+In this example, four different properties are defined; three use a *textbox* widget,
+while one uses a *dataset-selector*. The *groupByFields* property includes a :ref:`plugin
+function <plugins-presentation-plugin-function>`, used to fetch an output schema for the
+widget.
+
+
+.. _plugins-presentation-widgets:
+
+Plugin Widgets
+==============
+A widget in the CDAP UI represents a component that will be rendered and used to set the
+value of a property of a plugin. These are the different widgets |---| their type, their
+attributes (if any), their output data type, a description, sample JSON |---| that we support in
+CDAP pipelines as of version |version|:
+
+.. highlight:: json-ellipsis
+
+.. list-table::
```

Comment:
this table overflows horizontally:
![image](https://cloud.githubusercontent.com/assets/3895099/23172216/f0990860-f809-11e6-9b29-36c4111a52d7.png)